packages/extension/src/contexts/LedgerContext/LedgerContext.tsx

```diff
@@ -250,13 +262,41 @@ const LedgerProvider: FC = ({ children }) => {
     [client, getCurrentWallet]
   );
 
+  const getNFTData = useCallback(async ({ nft }: NftImageRequestPayload) => {
+    try {
+      const { URI } = nft;
+      let url = URI ? await convertHexToString(String(URI)) : '';
+
+      if (url.length === 0) {
+        throw new Error('URI is empty');
+      }
+
+      url = url.replace('ipfs://', 'https://ipfs.io/ipfs/');
+      const nftData = await fetch(url)
+        .then((res) => res.json())
+        .catch(() => ({
+          name: '-',
```

Comment:
Instead of putting a dash, maybe we could write something generic like: `No name`, or even better return null or undefined in order to don't display it in the UI?